### PR TITLE
Adapt the logging interface to work properly for our fact-based model.

### PR DESCRIPTION
**Pull request by geeknoid**
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/19)

### CODE REVIEW

File: api/v1/report.proto [hunk 1]
```diff
@@ -114,33 +114,32 @@ message LogEntry {
     EMERGENCY = 800;
   }
 
+  // The name of the LogEntryDescriptor providing the schema for this log entry.
+  string descriptor_name = 1;
+
   // Destination log for this entry
```
Comment (douglas-reid) [inline api/v1/report.proto]: not relevant to this review, but I wonder if we need to clarify terminology here a bit. something like:

"Destination collection for this log entry".

File: api/v1/report.proto [hunk 2]
```diff
@@ -114,33 +114,32 @@ message LogEntry {
     EMERGENCY = 800;
   }
 
+  // The name of the LogEntryDescriptor providing the schema for this log entry.
+  string descriptor_name = 1;
```
Comment (douglas-reid) [inline api/v1/report.proto]: checking: are these names fully qualified?  in other words, is this expected to be "service.example.io/logEntries/default" or just "default" (and proxy/mixer will be able to find the right descriptor)?

Comment (geeknoid) [inline api/v1/report.proto]: Depends on the case.

The metrics, quotas, and logs that the proxy will use will require well-known descriptor names. Those probably want to be qualified with something like proxy/<name>. For those descriptors for things that only the service itself will reference, such as for resource quotas or principals for permission checks, then those names are entirely under operator control and so don't need to be qualified.

File: api/v1/report.proto (hunk 1 above)
Comment (geeknoid) [inline api/v1/report.proto]: Will do

**Comment by geeknoid** [comment]
Updated PTAL.

**Pull request merged**